src/adapter/graphql/src/mutations/flows_mut/dataset_flow_configs_mut.rs

```diff
@@ -55,13 +53,18 @@ impl DatasetFlowConfigsMut {
         paused: bool,
         ingest: IngestConditionInput,
     ) -> Result<SetFlowConfigResult> {
-        if !ensure_set_config_flow_supported(dataset_flow_type, std::any::type_name::<Schedule>()) {
-            return Ok(SetFlowConfigResult::TypeIsNotSupported(
-                FlowTypeIsNotSupported,
-            ));
-        }
-        if let Some(e) =
-            ensure_expected_dataset_kind(ctx, &self.dataset_handle, dataset_flow_type).await?
+        let flow_run_config: FlowRunConfiguration = ingest.clone().into();
```

Comment:
I see that the input is later used in `ensure_expected_dataset_kind`, but maybe we should convert all inputs into domain types first and validate those?

I suspect all this logic will eventually migrate into a "use case" type where GQL types will be inaccessible.

Doing this will also avoid the need for all the `#derive(Clone)` you added everywhere.

Reply:
It is kinda tricky. 
Firstly, we may want to do some checks before validating user input. 
I've tried to go this way, but in this case we would need to cast domain `IngestRule` into `FlowConfigurationRule` which also will require `clone()`